File: backend/src/RepoFluent.Application/Lesson.cs
namespace RepoFluent.Application;

public sealed record Lesson(
    string Id,
    string Title,
    int EstimatedMinutes,
    string? Difficulty,
    IReadOnlyList<string>? Audience,
    IReadOnlyList<string>? Tags,
    bool IsRequired,
    int Order,
    IReadOnlyList<string>? Prerequisites,
    IReadOnlyList<string>? CompletionRules,
    IReadOnlyList<LearningObjective> Objectives,
    IReadOnlyList<Block> Blocks);

File: frontend/projects/api/src/lib/lesson.ts
import type { LessonBlock } from './lesson-block';
import type { LearningObjective } from './learning-objective';

export interface Lesson {
  id: string;
  title: string;
  estimatedMinutes: number;
  difficulty: string;
  audience: string[];
  tags: string[];
  isRequired: boolean;
  order: number;
  prerequisites: string[];
  completionRules: string[];
  objectives: LearningObjective[];
  blocks: LessonBlock[];
}
